### PR TITLE
fix for issue #741

### PR DESCRIPTION
**Pull request by jaswanthbjk**
The changes fixes the issue #741

### CODE REVIEW

File: torch_points3d/modules/MinkowskiEngine/api_modules.py [hunk 1]
```diff
@@ -74,11 +74,11 @@ def __init__(self, input_nc, output_nc, convolution, dimension=3):
             self.downsample = None
 
     def forward(self, x):
-        out = self.block(x)
+        intermediate_out = self.block(x)
```
Comment (CCInc) [inline torch_points3d/modules/MinkowskiEngine/api_modules.py:77]: I think you can leave the variables named `out` throughout all of this

File: torch_points3d/modules/MinkowskiEngine/api_modules.py [hunk 2]
```diff
@@ -201,12 +201,12 @@ def __init__(self, input_nc, output_nc, convolution, dimension=3, reduction=16):
         self.SE = SELayer(output_nc, reduction=reduction, dimension=dimension)
 
     def forward(self, x):
-        out = self.block(x)
-        out = self.SE(out)
+        intermediate_out_1 = self.block(x)
```
Comment (CCInc) [inline torch_points3d/modules/MinkowskiEngine/api_modules.py:204]: Same thing, I think these can remain as `out`

File: torch_points3d/modules/MinkowskiEngine/res16unet.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def forward(self, x):
         if self.downsample is not None:
             residual = self.downsample(x)
 
-        out += residual
+        out = out.clone() + residual
```
Comment (CCInc) [inline torch_points3d/modules/MinkowskiEngine/res16unet.py:49]: Do you know if the `.clone` is necessary? Will it work if you just do:

`out = out + residual`

**Review by CCInc — COMMENTED** [review]
Thanks so much for the PR!

I think there may also need to be some files changed in https://github.com/torch-points3d/torch-points3d/blob/master/torch_points3d/modules/MinkowskiEngine/resunet.py similar to in res16unet.py, would you mind adding the fix to that file as well?

**Comment by jaswanthbjk** [comment]
@CCInc I had done the changes specified and also some additional changes to SparseConv module

**Review by CCInc — APPROVED** [review]
Thanks so much!!

**Pull request merged**